src/components/shared/Slider/Marks.tsx

```diff
@@ -0,0 +1,354 @@
+import React, { FC, useEffect, useMemo } from 'react';
+import { StyleProp, StyleSheet, TouchableWithoutFeedback, ViewStyle } from 'react-native';
+
+import styled from 'styled-components/native';
+
+interface MarkPositionerType {
+  width: number;
+  position: number;
+}
+
+interface Props {
+  testID?: string;
+  sliderWidth: number;
+  style?: StyleProp<ViewStyle>;
+  mark?: React.ReactElement;
+  customMarkWidth?: number;
+  step?: number;
+  pixelsPerStep?: number;
+  markCount?: number;
+  startMark?: boolean;
+  endMark?: boolean;
+  disabled?: boolean;
+  onInit?: (markValues: number[], markPositions: number[]) => void | Promise<void>;
+  onMarkPress?: (value: number, position: number, index: number) => void | Promise<void>;
+}
+
+const DEFAULT = {
+  width: 3,
+  height: 3,
+};
+
+const Container = styled.View`
+  display: flex;
+  justify-content: center;
+  position: absolute;
+`;
+
+const MarkPositioner = styled.View<MarkPositionerType>`
+  position: absolute;
+  display: flex;
+  align-items: center;
+  width: ${({ width }): number => width};
+  left: ${({ position }): number => position};
+`;
+
+const Mark = styled.View`
+  width: ${DEFAULT.width};
+  height: ${DEFAULT.height};
+  background-color: #4163F4;
+`;
+
+// Checks if null or undefined. temporary until using othe library like 'lodash'.
+const isNil = (value: any): boolean => value == null;
+
+const getMarkCountByStep = ({
+  railWidth,
+  step,
+  startMark,
+  endMark,
+}: {
+  railWidth: number;
+  step: number;
+  startMark: boolean;
+  endMark: boolean;
+}): number => {
+  const count = Math.floor(railWidth / step);
```

Comment:
I remember we decided to handle `step` like Material UI API.
At the material UI the `step` is a value, not a distance. [https://material-ui.com/api/slider/](material)
I think the mark count should be `((maxValue - minValue) / step)`.
The user should not consider distance things to use Slider.

Reply:
I meant that `distance` is that `step` is the gap between two values even in material-ui.
https://material-ui.com/components/slider/

Reply:
Sorry for the broken link. I fixed it. Yeap, like you said the `step` is the gap between two marks. I mean that the unit of step should be `value` not a `distance`.

> The granularity with which the slider can step through values. (A "discrete" slider.) The min prop serves as the origin for the valid values. We recommend (max - min) to be evenly divisible by the step.
When step is null, the thumb can only be slid onto marks provided with the marks prop.

Reply:
Oh, yeah sorry for that. Then, should I change the word `distance` to `value`? or modify some logic? I didn't completely get the point of you mentioned. Sorry about that. Please a little more description for me. I might have miss understanding on that because of my weak English.

Reply:
@YongPilMoon Thank you so much for your review and consideration to merge it. I'll fix them in the next PR!

Reply:
No, you don't need to be sorry. I will try my best to explain it.
I think it's gonna help you to try Material UI Slider Demo.
<https://codesandbox.io/s/material-demo-yituy>
![step1](https://user-images.githubusercontent.com/22088158/72666186-35f27480-3a53-11ea-98c4-4a747af155d9.png)
We can see there are 11 marks. It's because we have to separate (maxValue - minValue) as 10 steps.
`marks = (110 - 10) / 10  + 1`
![step2](https://user-images.githubusercontent.com/22088158/72666190-3a1e9200-3a53-11ea-8f5f-f2bdade2eda2.png)
And when step is 20 there are 6 marks.
`marks = (110 - 10) / 20  + 1`

This code from material UI slider.
```
 const marks =
    marksProp === true && step !== null
      ? [...Array(Math.floor((max - min) / step) + 1)].map((_, index) => ({
          value: min + step * index,
        }))
      : marksProp;
```

But we are using distance to calculate marks count.
`const count = Math.floor(railWidth / step);`
It can make different mark count depend on railWidth even when step value is the same.